FMSgridtools/make_hgrid/hgridobj.py

```diff
@@ -0,0 +1,288 @@
+import dataclasses
+from typing import List, Optional
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from FMSgridtools.shared.gridtools_utils import check_file_is_there
+from FMSgridtools.shared.gridobj import GridObj
+
+class HGridObj():
+    def __init__(self):
+        self.tile = ""
+        self.x = None
+        self.y = None
+        self.dx = None
+        self.dy = None
+        self.area = None
+        self.angle_dx = None
+        self.angle_dy = None
+        self.arcx = ""
+
+    def write_out_hgrid(
+            self,
+            tilename,
+            outfile,
+            nx: int,
+            ny: int,
+            nxp: int,
+            nyp: int,
+            global_attrs: dict,
+            north_pole_tile="none",
+            north_pole_arcx="none",
+            projection="none",
+            geometry="none",
+            discretization="none",
+            conformal="none",
+            out_halo=0,
+            output_length_angle=0,
+    ):
+        tile = None
+        x = None
+        y = None
+        dx = None
+        dy = None
+        area = None
+        angle_dx = None
+        angle_dy = None
+        arcx = None
+        var_dict={}
+        if north_pole_tile == "none":
```

Comment:
the below code can probably be simplified to
```
 tile = xr.DataArray(  [tilename],
                attrs=dict(standard_name="grid_tile_spec",
                   geometry=geometry,
                   north_pole=north_pole_tile,
                  discretization=discretization,
                   conformal=conformal  ))
if north_pole_tile is not "none" : tile.tilename.assign_attrs(projection=projection)
```
something along that line.  It's less repetitious code